### ISSUE TITLE
Set up the CONTRIBUTING guide.

### ISSUE THREAD

**Issue by Aloento** [opentelekomcloud/gophertelekomcloud#390]
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

According to the discussion, we will move STYLEGUIDE to CONTRIBUTING and update the new version of the rules.

**Comment by Aloento** [comment]
As we will discard the `Extract`, I suggest that we change the file structure, `requests / results / urls`, from `results` to `types`. 
Then move all struct to `types.go` and classified by the method in which they are used.

Or let's write it in a better way.
Each file contains only one method with the resources it uses, and the file name is the same as the method name.

@anton-sidelnikov @outcatcher

**Comment by outcatcher** [comment]
Generally I'm against splitting files per method, but in this case it seems to be a better approach, both because files become smaller and because you don't have to move between files to get whole context.

**Comment by outcatcher** [comment]
@Aloento 
Generally I'm against splitting files per method, but in this case it seems to be a better approach, both because files become smaller and because you don't have to move between files to get whole context.

**Comment by Aloento** [comment]
@outcatcher yes, that's what I'm considering. We can get a better file structure, easier to find and organize.

**Comment by outcatcher** [comment]
Please add that using `fmt.Errorf("message: %w", err)` for wrapping errors is strongly encouraged. This improves understanding of errors.
